functions/Measure-DbaDiskSpaceRequirement.ps1

```diff
@@ -0,0 +1,355 @@
+            DB1             DB1             DB1_log   DB1_log     504   -1024   -520 INSTANCE2    D:\
+
+            ComputerName MountPoint RequiredSpaceKB
+            ------------ ---------- ---------------
+            INSTANCE2    D:\                   -448
+
+        .EXAMPLE
+            @([PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB1'},
+              [PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB2'}
+            ) | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a PSCustomObject with 2 databases to migrate
+
+        .EXAMPLE
+            Import-Csv -Path .\migration.csv -Delimiter "`t" | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a CSV file. You will need a header in migration.csv "Source<tab>Destination<tab>Database"
+
+        .EXAMPLE
+            Invoke-DbaSqlCmd -SqlInstance DBA -Database Migrations -Query 'select Source,Destination,DatabaseName from refresh.Migrations' `
+                | Measure-DbaDiskSpaceRequirement -Consolidate -Verbose
+
+            Using a SQL table. We are DBA after all!
+    #>
+
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SI','SqlInstance')]
+        [string]$Source,
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SD','Database','DatabaseName')]
+        [string]$SourceDatabase,
+        [Alias('SC')]
+        [PSCredential]$SourceSqlCredential,
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('DI')]
+        [string]$Destination, 
+        [Parameter(Mandatory=$false,ValueFromPipelineByPropertyName)]
+        [Alias('DD')]
+        [string]$DestinationDatabase,
+        [Alias('DC')]
+        [PSCredential]$DestinationSqlCredential,
+        [PSCredential]$Credential, # For Windows access to MountPoints
+        [switch]$Consolidate # Default would be detail only.
+    )
+    begin {
+        $NullText = '#NULL'
+        # TODO: What if multiple mountpoints exists on the same drive? Will $Path -like "$($M.Name)*" return the right one? I think we need to force a lazy RegEx
+
+        $local:CacheMP = @{}
+        $local:CacheDP = @{}
+
+        function Get-MountPointFromPath {
+            # Extract MountPoint from Path. This could be reuse I guess.
+            [CmdletBinding()]
+            Param(
+                [Parameter(Mandatory)]
+                $Path,
+                [Parameter(Mandatory)]
+                $ComputerName,
+                [PSCredential]$Credential
+            )
+            if(!$CacheMP[$ComputerName]) {
+                try {
+                    $CacheMP.Add($ComputerName, (Get-DbaDiskSpace -ComputerName $ComputerName -Credential $Credential -EnableException))
+                    Write-Verbose "CacheMP[$ComputerName] is in cache"
+                } catch {
+                    Write-Warning "Can't connect to $ComputerName. CacheMP[$ComputerName] = ?"
```

Comment:
This can error for other reasons than connection issues.

Reply:
As well, catch blocks need to utilize `Stop-Function` instead of a write command. You can view other commands in the module to see how it is used.

Reply:
In this case, I don't want to stop-function but to continue with a warning. So I guess Write-Message -Level Warning -Message should do?

Reply:
`Stop-Function .... -Continue`